packages/framework/src/Model/Cart/CartMigrationFacade.php

```diff
@@ -88,6 +88,11 @@ public function mergeCurrentCartWithCart(Cart $cart): void
      */
     public function onKernelController(ControllerEvent $event): void
     {
+        $controller = $event->getController();
+        $frontendApiControllerName = 'Shopsys\FrontendApiBundle\Controller\FrontendApiController';
+        if (class_exists($frontendApiControllerName) && isset($controller[0]) && get_class($controller[0]) === $frontendApiControllerName) {
```

Comment:
is the check for the existence of the class important here? 

Also `isset($controller[0])` will fail when someone creates an invokable controller (https://symfony.com/doc/current/controller/service.html#invokable-controllers) 

I suggest rewriting it like 

```php
if (is_array($controller)) {
    $controller = $controller[0];
}

if (get_class($controller) === $frontendApiControllerClass) {
```

Reply:
@grossmannmartin wouldn't `$controller = $controller[0]` fail if the array is empty?

Reply:
ok, I will rewrite it + check for the empty array, thanks! :+1:

Reply:
can be the array even empty?

Reply:
> is the check for the existence of the class important here?

as mentioned below, the API is not installed in the framework, project-base has the power to install and uninstall it so I can not be sure whether the class is available

Reply:
ah, ok, but now I see the `instanceof` does the trick :ok_hand: 33e5f73ca